src/Runner.Listener/Runner.cs

```diff
@@ -477,7 +478,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     // todo: add retries https://github.com/github/actions-broker/issues/49
                                     var runServer = HostContext.CreateService<IRunServer>();
                                     await runServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
-                                    var jobMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId);
+                                    Pipelines.AgentJobRequestMessage jobMessage = null;
+
+                                    var remainingRetryLimitTime = TimeSpan.FromMinutes(5);
+                                    while (true)
+                                    {
+                                        try
+                                        {
+                                            jobMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId);
+                                            break;
+                                        }
+                                        catch (Exception ex)
+                                        {
+                                            Trace.Error("Catch exception during get full job message");
+                                            Trace.Error(ex);
+
+                                            if (remainingRetryLimitTime > TimeSpan.Zero)
```

Comment:
Do we always want to retry? Should we break on 4xx response codes?

Reply:
I discussed it with @ericsciple, right now it is fine to retry all errors, but I keep thinking about non-retriable exceptions